cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -11,36 +11,100 @@
+    private static final double DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
 
     private Double diskMiB;
     private Integer cpuUtilization;
     private Double inboundNetworkKiBPerSecond;
     private Double outboundNetworkKiBPerSecond;
 
-    public Capacity(KafkaSpec spec) {
+    private int replicas;
+    private Storage storage;
+
+    public Capacity(KafkaSpec spec, Storage storage) {
         BrokerCapacity bc = spec.getCruiseControl().getBrokerCapacity();
 
-        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(spec.getKafka().getStorage());
+        this.replicas = spec.getKafka().getReplicas();
+        this.storage = storage;
+
+        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(storage);
         this.cpuUtilization = bc != null && bc.getCpuUtilization() != null ? bc.getCpuUtilization() : DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY;
         this.inboundNetworkKiBPerSecond = bc != null && bc.getInboundNetwork() != null ? getThroughputInKiB(bc.getInboundNetwork()) : DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
         this.outboundNetworkKiBPerSecond = bc != null && bc.getOutboundNetwork() != null ? getThroughputInKiB(bc.getOutboundNetwork()) : DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
     }
 
     /**
-     * Generate disk capacity configuration from the supplied storage configuration
+     * Generate broker capacity entry for capacity configuration.
+     *
+     * @param brokerId Id of broker
+     * @param diskCapacity Disk capacity configuration
+     * @param doc Documentation for broker entry
+     * @return Broker entry as a JsonObject
+     */
+    private JsonObject generateBrokerCapacity(int brokerId, JsonObject diskCapacity, String doc) {
+        JsonObject brokerCapacity = new JsonObject()
+            .put("brokerId", brokerId)
+            .put("capacity", new JsonObject()
+                .put("DISK", diskCapacity.getValue("DISK"))
+                .put("CPU", Integer.toString(cpuUtilization))
+                .put("NW_IN", Double.toString(inboundNetworkKiBPerSecond))
+                .put("NW_OUT", Double.toString(outboundNetworkKiBPerSecond))
+            )
+            .put("doc", doc);
+        return brokerCapacity;
+    }
+
+    /**
+     * Generate JBOD disk capacity configuration for a broker using the supplied storage configuration
      *
      * @param storage Storage configuration for Kafka cluster
-     * @return Disk capacity configuration value as a Double
+     * @param idx Index of the broker
+     * @return Disk capacity configuration value as a JsonObject for broker idx
+     */
+    private JsonObject generateJbodDiskCapacity(Storage storage, int idx) {
+        JsonObject json = new JsonObject();
+        String name = "";
```

Comment:
This could be defined directly in the loop below

Reply:
Should I move `path` down there as well? I thought it was more efficient to define the variables before the loop? Or does the compiler optimize make this not matter?

Reply:
I guess it the time saved (if any) is negligible , your suggestion makes it  more readable, I'll update it